extension.js

```diff
@@ -273,27 +272,45 @@ async function build(config, componentPath, server) {
 		}
 
 		// Build
-		const [command, ...args] = shellQuote.parse(config.buildCommand);
+		try {
+			logger.info(`Building Next.js application at ${componentPath}`);
+
+			const timerStart = performance.now();
+
+			const stdout = [];
+			const stderr = [];
 
-		const timerStart = performance.now();
+			const buildProcess = spawn(config.buildCommand, [], {
+				shell: true,
+				cwd: componentPath,
+				stdio: ['ignore', 'pipe', 'pipe'],
+			});
 
-		const { stdout, stderr, status, error } = spawnSync(command, args, {
-			cwd: componentPath,
-			encoding: 'utf-8',
-		});
+			buildProcess.stdout.on('data', c => stdout.push(c));
+			buildProcess.stderr.on('data', c => stderr.push(c));
+
+			const [code, signal] = await once(buildProcess, 'close');
 
-		if (status === 0) {
-			if (stdout) logger.info(stdout);
 			const duration = performance.now() - timerStart;
+
+			logger.info(`Build command \`${config.buildCommand}\` exited with code ${code} and signal ${signal}`);
```

Comment:
Maybe we could skip this if the (exit) code is 0 (there is already a log for starting and completion of the build)?

Reply:
sgtm